x-pack/legacy/plugins/siem/public/components/page/add_to_kql/index.tsx

```diff
@@ -5,33 +5,38 @@
  */
 
 import { EuiIcon, EuiPanel, EuiToolTip } from '@elastic/eui';
-import { isEmpty } from 'lodash/fp';
+import { Filter } from '@kbn/es-query';
+import { getOr } from 'lodash/fp';
 import React from 'react';
+import { connect } from 'react-redux';
 import styled from 'styled-components';
 import { StaticIndexPattern } from 'ui/index_patterns';
+import { Query } from 'src/plugins/data/common';
 
-import { HostsFilter } from '../../../containers/hosts';
-import { NetworkFilter } from '../../../containers/network';
-import { assertUnreachable } from '../../../lib/helpers';
-import { hostsModel, KueryFilterQuery, networkModel } from '../../../store';
 import { WithHoverActions } from '../../with_hover_actions';
+import { InputsModelId } from '../../../store/inputs/constants';
+import { siemFilterManager } from '../../search_bar';
+
 import * as i18n from './translations';
+import { filterQuerySelector } from '../../search_bar/selectors';
+import { State } from '../../../store';
+import { InputsRange } from '../../../store/inputs/model';
+
+interface AddToKqlRedux {
+  query: Query;
+}
 
-interface Props {
-  applyFilterQueryFromKueryExpression: (expression: string) => void;
+interface OwnProps {
+  id: InputsModelId;
   children: JSX.Element;
-  expression: string;
-  filterQueryDraft: KueryFilterQuery;
+  indexPattern: StaticIndexPattern;
+  filter: Filter;
 }
 
-const AddToKqlComponent = React.memo<Props>(
-  ({ children, expression, filterQueryDraft, applyFilterQueryFromKueryExpression }) => {
+const AddToKqlComponent = React.memo<OwnProps & AddToKqlRedux>(
+  ({ children, id, indexPattern, filter, query }) => {
     const addToKql = () => {
-      applyFilterQueryFromKueryExpression(
-        filterQueryDraft && !isEmpty(filterQueryDraft.expression)
-          ? `${filterQueryDraft.expression} and ${expression}`
-          : expression
-      );
+      siemFilterManager.addFilters(filter);
```

Comment:
Seems our good friend `AddToKqlComponent` really benefited from this -- so simple now! Nice!